utm_tracker/session.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from typing import Any, List
+
+from django.contrib.sessions.backends.base import SessionBase
+
+from .models import LeadSource
+from .types import UtmParamsDict
+
+SESSION_KEY_UTM_PARAMS = "utm_params"
+
+logger = logging.getLogger(__name__)
+
+
+def stash_utm_params(session: SessionBase, params: UtmParamsDict) -> bool:
+    """
+    Add a new utm_params dictionary to the session.
+
+    If the params dict is empty ({}), then it is ignored.
+
+    Returns True if the params are stored, else False.
+
+    """
+    if params:
+        session.setdefault(SESSION_KEY_UTM_PARAMS, [])
+        session[SESSION_KEY_UTM_PARAMS].append(params)
+        return True
+    return False
+
+
+def pop_utm_params(session: SessionBase) -> List[UtmParamsDict]:
+    """Pop the list of utm_param dicts from a session."""
+    return session.pop(SESSION_KEY_UTM_PARAMS, [])
+
+
+def flush_utm_params(user: Any, session: SessionBase) -> List[LeadSource]:
```

Comment:
s/flush/persist? or s/flush/dump?

"flush" to me as a tech term means "to get rid of" (in particular for caches, redis-cli's flush_all for instance) or to abord/delete. While it is clearing the session - it's not losing any data, as it's persisting the leads to disk so this was confusing to me until I read it.